From cd66b934f392b7fbf7b2cf57ff4b47d1c085b07d Mon Sep 17 00:00:00 2001
From: "Nguyen Van Tu [FGA.AIS]" <TuNV16@fsoft.com.vn>
Date: Fri, 2 Nov 2018 17:57:08 +0700
Subject: [PATCH 027/628] linux-renesas:r8a7745-dtsi:add vcp0 vpc0 prr vpc1
 vpc0xy vpc1xy for uvcs-kernel-module to support LTS4.4.6

Reference to commit "linux-renesas:r8a7743-dtsi:add vcp0 vpc0 prr vpc1 vpc0xy vpc1xy for uvcs-kernel-module to support LTS4.4.6"

In details:
These are issues when porting from 3.10 to 4.4
    Issue 1: Can not request interrupt for HW IP (VCP)
    Issue 2: Can not map HW IP register (VPC)
These is a difference in register platform device for the driver:
    - The old driver use kernel API (platform_device_register())
to register it device to kernel,
    - The new driver, device binding to driver is register by the new way
that use the device info from <target-board>.dtb file.

vpc1, vpc0xy, and vpc1xy are added as reserve only vcp0, vpc0 and prr is used.
The prr is used for getting LSI information which tell the driver
what kind of board it running

Signed-off-by: Long Dang <long.dang.aj@rvc.renesas.com>
Signed-off-by: Nguyen Van Linh [FGA.BU13] <LinhNV25@fsoft.com.vn>
Signed-off-by: vietn <vietn@fsoft.com.vn>
Signed-off-by: Nguyen Van Tu [FGA.AIS] <TuNV16@fsoft.com.vn>
---
 arch/arm/boot/dts/r8a7745.dtsi | 48 ++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 48 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7745.dtsi b/arch/arm/boot/dts/r8a7745.dtsi
index 8e4c1bf..dfdef01 100644
--- a/arch/arm/boot/dts/r8a7745.dtsi
+++ b/arch/arm/boot/dts/r8a7745.dtsi
@@ -923,6 +923,54 @@
 			status = "disabled";
 		};
 
+		vcp0: vcp0@fe900000 {
+			compatible = "renesas,vcp0";
+			reg = <0 0xfe900000 0 0x200>, <0 0xfe900200 0 0x200>;
+			interrupts = <0 258 IRQ_TYPE_LEVEL_HIGH>,
+				<0 259 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mstp1_clks R8A7745_CLK_VCP0>;
+			power-domains = <&cpg_clocks>;
+			renesas,#ch = <0>;
+		};
+
+		vpc0: vcp0@fe908000 {
+			compatible = "renesas,vpc0";
+			reg = <0 0xfe908000 0 0x90>;
+			clocks = <&mstp1_clks R8A7745_CLK_VPC0>;
+			power-domains = <&cpg_clocks>;
+			renesas,#ch = <0>;
+		};
+
+		vpc1: vcp0@fe918000 {
+			compatible = "renesas,vpc1";
+			reg = <0 0xfe918000 0 0x90>;
+			/* clocks = <&mstp1_clks R8A7745_CLK_VPC1>; */
+			power-domains = <&cpg_clocks>;
+			renesas,#ch = <0>;
+		};
+
+		vpc0xy: vcp0@fe960380 {
+			compatible = "renesas,vpc0xy";
+			reg = <0 0xfe960380 0 0x4>;
+			/* clocks = <&mstp1_clks R8A7745_CLK_VPC0>; */
+			power-domains = <&cpg_clocks>;
+			renesas,#ch = <0>;
+		};
+
+		vpc1xy: vcp0@fe960384 {
+			compatible = "renesas,vpc1xy";
+			reg = <0 0xfe960384 0 0x4>;
+			/* #clocks = <&mstp1_clks R8A7745_CLK_VPC1>; */
+			power-domains = <&cpg_clocks>;
+			renesas,#ch = <0>;
+		};
+
+		prr: vcp0@ff000044 {
+			compatible = "renesas,prr";
+			reg = <0 0xff000044 0 0x4>;
+			renesas,#ch = <0>;
+		};
+
 		vin0: video@e6ef0000 {
 			compatible = "renesas,vin-r8a7745",
 				     "renesas,rcar-gen2-vin";
-- 
2.7.4

